src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,259 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from multiprocessing import Queue
+import threading
+
+from avocado.core.exceptions import TestFail
+from dmg_utils import get_storage_query_device_info, get_dmg_response
+from exception_utils import CommandFailure
+from ior_utils import run_ior, thread_run_ior
+from job_manager_utils import get_job_manager
+from led import VmdLedStatus
+from nvme_utils import set_device_faulty
+
+
+class NvmeFaultReintegrate(VmdLedStatus):
+    """
+    Test Class Description: This test runs
+    NVME fault and reintegration test cases.
+
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-ancestors
+    def setUp(self):
+        """Set up for test case."""
+        super().setUp()
+        self.daos_command = self.get_daos_command()
+
+    def verify_dev_led_state(self, device, dev_state="NORMAL", led_state="OFF"):
+        """Verify device dev_state and led_state.
+
+        Args:
+            device (str): the device description.
+            dev_state (str): expect dev_state, default to "NORMAL".
+            led_state (str): expect led_state, default to "OFF".
+
+        Returns:
+            bool: True if the expected dev_state and led_state.
+
+        """
+        return self.check_result(self.get_led_status_value(device), dev_state, led_state)
+
+    def check_result(self, result, dev_state, led_state):
+        """Check for result of storage device and led states.
+
+        Args:
+            result (dict): return from get_led_status_value for a storage device.
+            dev_state (str): expect dev_state.
+            led_state (str): expect led_state.
+
+        Return:
+            True if the expected dev_state and led_state.
+
+        """
+        if "response" in result.keys():
+            for value in list(result['response']['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        elif "host_storage_map" in result.keys():
+            for value in list(result['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        else:
+            self.log.debug(
+                "Neither 'response' nor 'host_storage_map' found in dmg command json output:")
+            self.log.debug('  %s', result)
+        return False
```

Comment:
Now that all of the `result` inputs to this method are standardized, this can be simplified down to:
```suggestion
        if "response" in result.keys():
            for value in list(result['response']['host_storage_map'].values()):
                if value['storage']['smd_info']['devices']:
                    for device in value['storage']['smd_info']['devices']:
                        self.log.debug(
                            'Verifying: dev_state (%s == %s) and led_state (%s == %s)',
                            device['dev_state'], dev_state, device['led_state'], led_state)
                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
                            return True
        return False
```

Reply:
Above scripts are handling the response difference for both methods, response with and without key "response"

Reply:
Correction:  the input to `check_result()` should be the output from `get_dmg_response` - essentially the dmg json ['response'] value.  This means that the dit should now look like this:
```
        for value in list(result['host_storage_map'].values()):
            if value['storage']['smd_info']['devices']:
                for device in value['storage']['smd_info']['devices']:
                    self.log.debug(
                        'Verifying: dev_state (%s == %s) and led_state (%s == %s)',
                        device['dev_state'], dev_state, device['led_state'], led_state)
                    if device['dev_state'] == dev_state and device['led_state'] == led_state:
                        return True
        return False
```

And then we need to replace https://github.com/daos-stack/daos/blob/ad7c932cf1a2c223290e194018cd756b57896f55/src/tests/ftest/vmd/fault_reintegration.py#L233 with:
```
        result = get_dmg_response(test, self.dmg.storage_replace_nvme, old_uuid=test_dev, new_uuid=test_dev)
```

Reply:
done.